cli/command/service/opts.go

```diff
@@ -479,6 +481,7 @@ func addServiceFlags(cmd *cobra.Command, opts *serviceOptions) {
 	flags.BoolVar(&opts.healthcheck.noHealthcheck, flagNoHealthcheck, false, "Disable any container-specified HEALTHCHECK")
 
 	flags.BoolVarP(&opts.tty, flagTTY, "t", false, "Allocate a pseudo-TTY")
+	flags.BoolVar(&opts.readonlyRootfs, flagReadonlyRootfs, false, "Mount the container's root filesystem as read only")
```

Comment:
Oh, can you version this flag, so that it only shows if the daemon is docker 1.14 or above?

```go
flags.SetAnnotation(flagReadonlyRootfs, "version", []string{"1.26"})
```

(hm, thinking of which, I think there's some other flags that we need to update as well)